### DISCUSSION TITLE
bitbybit-occt and bitbybit-occt-worker npm libraries

### DISCUSSION THREAD

**Discussion by matas-bitbybit-dev** [donalffons/opencascade.js#233] [Show and tell]
Hi @donalffons,

Just wanted to let you and the community know that I separated some OCCT code of [bitbybit.dev](https://bitbybit.dev) into separate npm packages:

NPM
[bitbybit-occt](https://www.npmjs.com/package/@bitbybit-dev/occt)
[bitbybit-occt-worker](https://www.npmjs.com/package/@bitbybit-dev/occt-worker)

GITHUB
[bitbybit-occt](https://github.com/bitbybit-dev/bitbybit-occt)
[bitbybit-occt-worker](https://github.com/bitbybit-dev/bitbybit-occt-worker)

**bitbybit-occt** uses custom build of opencascade.js and is just a layer on top of opencascade.js, thus it can be used in node and browser environments
**bitbybit-occt-worker** is worker implementation with some basic caching and async interface to the bitbybit-occt API

These packages can be used by people to find some of the basic algorithms that took me a while to figure out how to write with OCCT. Also I took some steps to take care of the basic memory management. The codebase is still a bit messy, but it works. I'm also putting in some time to unit test that codebase.

I also open sourced some basic example apps here:
[App Examples](https://github.com/bitbybit-dev/app-examples)

These example apps use packages mentioned before in Angular, React, basic Node and Express apps. There are examples for both BabylonJS and ThreeJS. Some of the apps use [@bitbybit-dev/core ](https://www.npmjs.com/package/@bitbybit-dev/core) package which is a layer above occt and other libraries I integrate.

Having said that [bitbybit.dev](https://bitbybit.dev) app calls exactly the same code through @bitbybit-dev/core package, but it uses various visual programming editors. I hope that this will allow people learn the algorithms and make it possible to apply the code in different apps and environments.

**Few questions:**

@bitbybit-dev/occt packages are shared under MIT license, I see that you applied different LGPL license and I wonder if that implies that other projects should also use LGPL which use OCCT?

Question about the future. I wonder what your plans are for this project and if you plan to update opencascade.js with the latest versions of OCCT.

Thanks for your efforts that made all of this happen!

**Comment by cesarecaoduro** [comment]
@donalffons & @matas-bitbybit-dev 

I have also built a quick starter prototype using Nuxt3 ;)
[GitHub repository](https://github.com/cesarecaoduro/nuxt3-occt)

**Comment by cesarecaoduro** [comment]
@matas-bitbybit-dev  I know you have done a lot of work in this space, any idea on how to help on this topic #280 ?
Also have you planned to have an example of apps using ```Vue```?

**Reply by matas-bitbybit-dev** [reply]
Hey @cesarecaoduro, I replied in the topic that you linked. I worked with Vue very sparadically, so I did not feel confident to provide an example, but I guess I could try if that would be helpful for you.

**Reply by matas-bitbybit-dev** [reply]
If someone is looking for a nuxt 3 example, here it is:

 https://github.com/bitbybit-dev/app-examples/tree/main/nuxt/basic
